### PR TITLE
EMF_ECS integ test

### PR DESCRIPTION
**Pull request by dchappa**
# Description of the issue
We lacked integration testing for publishing metrics via EMF on ECS clusters

# Description of changes
This test will ensure that testing for ECS clusters that have the agent publishing via EMF will publish fine.

# License
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

# Tests

### CODE REVIEW

File: test/metric/dimension/emf_ecs_dimension_provider.go
Comment (dchappa) [reply test/metric/dimension/emf_ecs_dimension_provider.go]: This function is called in the provider.go file, and is required to implement the interface. I was tinkering with the dimensions a little bit, will put up a revision that actually checks for the instance name

File: test/emf_ecs/emf_ecs_test.go
Comment (dchappa) [reply test/emf_ecs/emf_ecs_test.go:83]: Got it, I do remember seeing some open PR's that make it a common function, and I'm sure that if it gets merged in I'll have to tinker with this code block. Just implemented it to unblock myself

File: test/metric/dimension/emf_ecs_dimension_provider.go
Comment (dchappa) [reply test/metric/dimension/emf_ecs_dimension_provider.go]: fixed in latest commit

File: test/emf_ecs/ecs_resources/ec2_launch/daemon/ecs_taskdef.tpl
Comment (SaxyPandaBear) [inline test/emf_ecs/ecs_resources/ec2_launch/daemon/ecs_taskdef.tpl:77]: Ah okay that explains it. Nitpick, but I think that using the `InstanceID` dimension and hardcoding `INSTANCEID` is kind of convoluted and doesn't actually use the instance ID for the instance running. I'd rather just drop an arbitrary dimension here.

**Review by SaxyPandaBear — APPROVED** [review]
lgtm

File: test/emf_ecs/emf_ecs_test.go
Comment (SaxyPandaBear) [reply test/emf_ecs/emf_ecs_test.go:83]: https://github.com/aws/amazon-cloudwatch-agent-test/blob/88b1a960e56528436473416a7040b06375917e64/test/metric/metric_validation_util.go#L8 looks to be where it landed after the other PRs got merged

Comment (dchappa) [reply test/emf_ecs/emf_ecs_test.go]: fixed in latest commit

**Review by SaxyPandaBear — APPROVED** [review]

File: test/emf_ecs/ecs_resources/ec2_launch/daemon/ecs_taskdef.tpl
Comment (sky333999) [inline test/emf_ecs/ecs_resources/ec2_launch/daemon/ecs_taskdef.tpl:1]: These probably could all be nested as `emf` dir under the existing `ecs` dir instead of creating a new root level dir..
But I think a lot of the structure in this repo needs to change by a lot anyway - so not going to block on it for now.

**Pull request merged**
